src/Models/Timeline.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Plank\Checkpoint\Models;
+
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+
+/**
+ * @property-read Collection|Checkpoint[] $checkpoints
+ */
+class Timeline extends Model
+{
+    /**
+     * Get all checkpoints associated with the Timeline
+     *
+     * @return HasMany
+     */
+    public function checkpoints(): HasMany
+    {
+        /** @var Checkpoint|string $checkpointClass */
+        $checkpointClass = config('checkpoint.models.checkpoint');
+
+        return $this->hasMany($checkpointClass, $checkpointClass::TIMELINE_ID);
+    }
+}
```

Comment:
Are we also maybe missing a `setActive()` on timelines as well? i see that works pretty well with checkpoints. In this use case are users just expected to filter on timelines themselves?